cycle2/empanelment/cycle2_empanelment_vs_prep.r

```diff
@@ -0,0 +1,119 @@
+# script to process cycle2 empanelment data for volunteer science upload
+
+# libraries and constants
+library(dplyr)
+library(hot.deck)
+
+complete_vars <- c(
+    'RecipientFirstName',
+    'RecipientLastName',
+    'RecipientEmail',
+    'ResponseID',
+    'Q40',
+    'Q57_1',
+    'Q57_2',
+    'Q57_3',
+    'Q57_4',
+    'Q57_5',
+    'Q57_6',
+    'Q57_7',
+    'Q57_8',
+    'Q57_9',
+    'Q57_10',
+    'Q57_11',
+    'Q57_12',
+    'Q107_1',
+    'Q107_2',
+    'Q107_3',
+    'Q107_4',
+    'Q107_5',
+    'Q107_6',
+    'Q107_8'
+)
+imputation_variables <- 'nid|Q107|Q1[1-8]$|Q2[2-7]|Q29|Q56|Q57|Q59|Q6|Q7[0-9]|Q8[0-9]|q9[0-9]'
+
+# define functions
+ambiguity_calc <- function(x) {
+    revs <- c(1, 4, 5, 9, 10, 11, 12)
+    x[, paste0('Q57_', revs)] <- apply(x[, paste0('Q57_', revs)], 2, function(y) {
+        return(abs(y - 5) + 1)
+    })
+    return(apply(x[, grep('Q57_', names(x))], 1, function(y) {sum(y)}))
+}
+
+randomize_data <- function(x) {
+    half <- floor(nrow(x) / 2)
+    x <- x[order(x$tol_ambiguity), ]
+    lh <- x[1:half, ]
+    uh <- x[(half + 1):nrow(x), ]
+    lh_switch <- sample(1:nrow(lh), floor(half * .2))
+    uh_switch <- sample(1:nrow(uh), floor(half * .2))
+    lh <- rbind(lh, uh[uh_switch, ])
+    uh <- rbind(uh, lh[lh_switch, ])
+    lh <- lh[-lh_switch, ]
+    uh <- uh[-uh_switch, ]
+    return(list('low_tol' = lh, 'high_tol' = uh))
+}
+
+trans_leadership <- function(x, choice=c('sum', 'mean')) {
+    if(choice == 'sum') {
+        tmp <- apply(x[, grep('Q107_', names(x))], 1, sum)
+    } else {
+        tmp <- apply(x[, grep('Q107_', names(x))], 1, mean)
+    }
+    return(tmp)
+}
+
+# instructions
```

Comment:
I would move these instructions to the top, I was confused for example by the missing 'Q107_7' variable until I read this comment.